docs/docsite/rst/network/getting_started/network_differences.rst

```diff
@@ -0,0 +1,72 @@
+
+Because network modules execute on the control node instead of on the managed nodes, they can support multiple communication protocols. The communication protocol (XML over SSH, CLI over SSH, API over HTTPS) selected for each network module depends on the platform and the purpose of the module. Some network modules support only one protocol; some offer a choice. The most common protocol is CLI over SSH. You set the communication protocol with the ``ansible_connection`` variable:
+
+.. csv-table::
+   :header: "Value of ansible_connection", "Protocol", "Requires"
+   :widths: 30, 10, 10
+
+   "network_cli", "CLI over SSH", "network_os setting"
+   "netconf", "XML over SSH", "network_os setting"
+   "local", "depends on provider", "provider setting"
+
+Beginning with Ansible 2.5, we recommend using ``network_cli`` or ``netconf`` for ``ansible_connection`` whenever possible. For details on using API over HTTPS connections, see the platform-specific pages.
+
+Modules Organized by Network Platform
+================================================================================
+
+A network platform is a set of network devices with a common operating system that can be managed by a collection of modules.  The modules for each network platform share a prefix, for example: 
+
+- Arista: ``eos_``
+- Cisco: ``ios_``, ``iosxr_``, ``nxos_``
+- Juniper: ``junos_``
+- VyOS ``vyos_``
+
+All modules within a network platform share certain requirements. Some network platforms have specific differences - see the platform-specific documentation for details.
+
+Privilege Escalation: `authorize` and `become`
+================================================================================
+
+Several network platforms support privilege escalation, where certain tasks must be done by a privileged user. This is generally known as ``enable`` mode (the equivalent of ``sudo`` in \*nix administration). Ansible network modules offer privilege escalation for those network devices that support it. However, different platforms use privilege escalation in different ways. 
+
+Network platforms that support ``connection: network_cli`` and privilege escalation use the Ansible parameter ``become: yes`` with ``become_method: enable``. For modules in these platforms, a ``group_vars`` file would look like:
+
+.. code-block:: yaml
+
+   ansible_connection: network_cli
+   ansible_network_os: ios
+   ansible_become: yes
+   ansible_become_method: enable
+
+We recommend using ``network_cli`` connections whenever possible. However, in Ansible < 2.5, some network platforms support privilege escalation but not ``network_cli`` connections. In that case, you must use ``authorize: yes`` and ``auth_pass: my_enable_password`` in the ``provider`` settings. For that use case, a ``group_vars`` file would look like:
+
+.. code-block:: yaml
+
+   ansible_connection: local
+   ansible_network_os: eos
+   provider:
+     authorize: yes
+     auth_pass: " {{ secret_auth_pass }}"
+     transport: eapi
```

Comment:
may be add some more info about other options:
```
eapi:
  transport: eapi
  use_ssl: no
  port: 80
  authorize: yes
```
Then add an example on how to use it:
```
  tasks:
    - name: provider demo with eos
      eos_banner:
        banner: motd
        text: |
          this is test
          of multiline
          string
        state: present
        provider: "{{ eapi }}"
```

Note: This is supported with `connection: local` in ansible < 2.5 and both `connection(local, network_cli)` in 2.5.

Reply:
Thanks for the detailed example.